getter.go

```diff
@@ -132,11 +138,15 @@ func (g *getter) queueFile(url *url.URL) (http.Header, error) {
 
 	g.chunkWg.Add(1)
 	resp, err := g.retryRequest("GET", url.String(), nil)
+
+	// resp could be nil, depending on the error.
 	if err != nil {
-		return resp.Header, err
+		logger.debugPrintf("ERROR on queueFile", errgo.Mask(err))
+		return nil, err
 	}
 
```

Comment:
We should probably add a check here for nil resp before we start trying to access the resp;    that would give us the 3 conditions:
err != nil then return and error
resp == nil then return and error
resp.StatusCode != 200 then return and error

I believe the contract on the interface says you would get back an err if the resp is nil, but we can go ahead and add the check and not assume